tapo@smiklosovic/applet.js

```diff
@@ -0,0 +1,289 @@
+            this.opt.enabled = state;
+            this.doGlobalToggle();
+        }.bind(this));
+
+        this.menu.addMenuItem(this.enabledDay);
+
+        for (let device of Object.keys(this.opt.devices)) {
+
+            deviceSwitches[device] = new QPopupSwitch({
+                label: _(device),
+                active: this.opt.devices[device]
+            });
+            deviceSwitches[device].connect('toggled', function(item, state) {
+                this.opt.devices[device] = state;
+                this.doUpdate();
+            }.bind(this));
+            this.menu.addMenuItem(deviceSwitches[device]);
+        }
+    }
+
+    doGlobalToggle() {
+        if (this.opt.enabled) {
+            for (let device of Object.keys(this.opt.devices)) {
+                this.toggleDevice(device, 'on');
+                this.opt.devices[device] = true;
+                deviceSwitches[device].setToggleState(true);
+            }
+        } else {
+            for (let device of Object.keys(this.opt.devices)) {
+                this.toggleDevice(device, 'off');
+                this.opt.devices[device] = false;
+                deviceSwitches[device].setToggleState(false);
+            }
+        }
+    }
+
+    doUpdate() {
+        Object.keys(this.opt.devices).forEach(key => {
+            if (this.opt.devices[key]) {
+                this.toggleDevice(key, 'on');
+            } else {
+                this.toggleDevice(key, 'off');
+            }
+        });
+    }
+
+    // device interaction
+
+    toggleDevice(device_name, state) {
+        GLib.spawn_command_line_async("/bin/bash -c '" + TOGGLE_DEVICE_SCRIPT + " " + device_name + " " + state + " " + this.opt.serviceUrl + " " + this.opt.authToken + "'");
+    }
+
+    getDeviceState(device_name) {
+        let [a, stdout, c, d] = GLib.spawn_command_line_sync("/bin/bash -c '" + TOGGLE_DEVICE_SCRIPT + " " + device_name + " status " + this.opt.serviceUrl + " " + this.opt.authToken + "'");
+        return stdout.toString();
+    }
+
+    // applet specific
+
+    on_applet_clicked(event) {
+        this.statusCheck();
+        this.menu.toggle();
+    }
+
+    updateIcon() {
+        if (this._icon) {
+            this._icon.destroy();
+        }
+
+        let iconSize = this.getPanelIconSize(St.IconType.SYMBOLIC);
+        this._icon = new St.Icon({
+            icon_name: "weather-clear-night",
+            icon_type: St.IconType.SYMBOLIC,
+            reactive: true,
+            track_hover: true,
+            style_class: 'applet-icon'
+        });
+
+        this._iconBin.set_child(this._icon);
+        this.on_panel_icon_size_changed()
+    }
+
+    on_panel_icon_size_changed() {
+        let iconSize = this.getPanelIconSize(St.IconType.SYMBOLIC);
+        this._iconBox.set_size(iconSize, iconSize);
+        this._icon.set_icon_size(iconSize);
+    }
+
+    reloadApplet() {
+        let cmd = `/usr/bin/cinnamon-dbus-command ReloadXlet ${this.metadata.uuid} APPLET`;
```

Comment:
```suggestion
        let cmd = `/usr/bin/cinnamon-dbus-command ReloadXlet '${this.metadata.uuid}' APPLET`;
```

Sorry, this needs quotes to function properly.